### PR TITLE
Harden panel/section UI code by removing contents from being logically nested (#34391)

### PR DESCRIPTION
**Pull request by delawski**

### CODE REVIEW

File: src/wp-admin/js/customize-controls.js [hunk 1]
```diff
@@ -188,6 +186,32 @@
 	};
 
 	/**
+	 * Return browser supported `transitionend` event name
+	 *
+	 * @since
```
Comment (westonruter) [inline src/wp-admin/js/customize-controls.js]: @delawski this can be `@since 4.7.0`

**Comment by westonruter** [comment]
Some JSCS issues which may or may not be relevant:

```
src/wp-admin/js/customize-nav-menus.js: line 1038, col 23, disallowSpacesInAnonymousFunctionExpression: Illegal space before opening round brace
src/wp-admin/js/customize-nav-menus.js: line 1038, col 23, disallowSpacesInFunctionExpression: Illegal space before opening round brace
src/wp-admin/js/customize-nav-menus.js: line 2296, col 4, requireVarDeclFirst: Variable declarations must be the first statements of a function scope.
src/wp-admin/js/nav-menu.js: line 86, col 8, requireSpacesInsideParentheses: Missing space after opening round bracket
src/wp-admin/js/nav-menu.js: line 86, col 12, requireSpacesInsideParentheses: Missing space before closing round bracket
src/wp-admin/js/nav-menu.js: line 86, col 26, requireSpacesInsideParentheses: Missing space after opening round bracket
src/wp-admin/js/nav-menu.js: line 86, col 44, requireSpaceBeforeBinaryOperators: Operator + should not stick to preceding expression
src/wp-admin/js/nav-menu.js: line 87, col 17, requireSpacesInsideParentheses: Missing space after opening round bracket
src/wp-admin/js/nav-menu.js: line 87, col 35, requireSpaceBeforeBinaryOperators: Operator + should not stick to preceding expression
src/wp-admin/js/nav-menu.js: line 87, col 38, requireSpacesInsideParentheses: Missing space after opening round bracket
src/wp-admin/js/nav-menu.js: line 87, col 52, requireSpacesInsideParentheses: Missing space before closing round bracket
```

File: src/wp-admin/js/customize-controls.js [hunk 2]
```diff
@@ -188,6 +186,32 @@
 	};
 
 	/**
+	 * Return browser supported `transitionend` event name
```
Comment (westonruter) [inline src/wp-admin/js/customize-controls.js]: Add period to end.

File: src/wp-admin/js/customize-controls.js [hunk 3]
```diff
@@ -188,6 +186,32 @@
...
+	 *
+	 * @since 
+	 *
+	 * @returns {String}
```
Comment (westonruter) [inline src/wp-admin/js/customize-controls.js]: Good to add a description for the return, like "Normalized transitioned event name."

File: src/wp-admin/js/customize-controls.js [hunk 4]
```diff
@@ -483,6 +508,63 @@
 		},
 
 		/**
+		 * Animate container state change if transitions are supported by the browser.
+		 *
+		 * @param {function} completeCallback Function to be called after transition is completed.
+		 * @private
```
Comment (westonruter) [inline src/wp-admin/js/customize-controls.js]: Add `@returns {void}`

File: src/wp-admin/js/customize-controls.js [hunk 5]
```diff
@@ -483,6 +508,63 @@
...
+
+			var construct = this,
+				content = construct.contentContainer,
+				overlay = content.closest( '.wp-full-overlay' ),
+				elements, transitionEndCallback;
+
+			// Determine set of elements that are affected by the animation.
+			elements = overlay.add( content );
+			if ( _.isUndefined( construct.panel ) || '' === construct.panel() ) {
+					elements = elements.add( '#customize-info, .customize-pane-parent' );
```
Comment (westonruter) [inline src/wp-admin/js/customize-controls.js]: Looks like indentation may be off here.

File: src/wp-admin/js/customize-nav-menus.js [hunk 6]
```diff
@@ -1028,6 +1028,15 @@
 				content.slideUp( 'fast' );
 				content.find( '.menu-name-field' ).removeClass( 'invalid' );
 			}
+		},
+
+		/**
+		 * Detach and return the content html, extracted from the container html, if it exists.
+		 *
+		 * @since
```
Comment (westonruter) [inline src/wp-admin/js/customize-nav-menus.js]: Add `@since 4.7.0`

Comment (westonruter) [inline src/wp-admin/js/customize-nav-menus.js]: Also add `@returns {jQuery} Content UL element.`

**Comment by delawski** [comment]
@westonruter - I've fixed all the issues you've mentioned. The JSCS errors are not relevant, as I haven't touched those parts of the code.

**Comment by westonruter** [comment]
Committed to `trunk` in https://core.trac.wordpress.org/changeset/38648

Great work, @delawski! 🎉

**Pull request closed without merging**